source/server/connection_handler_impl.h

```diff
@@ -93,7 +94,11 @@ class ConnectionHandlerImpl : public Network::TcpConnectionHandler,
   const absl::optional<uint32_t> worker_index_;
   Event::Dispatcher& dispatcher_;
   const std::string per_handler_stat_prefix_;
-  std::list<std::pair<Network::Address::InstanceConstSharedPtr, ActiveListenerDetails>> listeners_;
+
+  absl::flat_hash_map<uint64_t, std::vector<std::shared_ptr<ActiveListenerDetails>>>
+      listener_map_by_tag_;
+  absl::flat_hash_map<std::string, std::shared_ptr<ActiveListenerDetails>> listener_map_by_address_;
```

Comment:
is the map string really ipaddress and port?

Reply:
Yes, it gets from `Address::Instance`, here is the ipv6 https://github.com/envoyproxy/envoy/blob/d3ef83a5cb822e0ba1a13e4c32abf6f061bdddf1/source/common/network/address_impl.cc#L296

and ipv4 https://github.com/envoyproxy/envoy/blob/d3ef83a5cb822e0ba1a13e4c32abf6f061bdddf1/source/common/network/address_impl.cc#L141

Reply:
also, I separated part of refactoring into another pr https://github.com/envoyproxy/envoy/pull/19362